Cabal/Distribution/Simple/PackageIndex.hs

```diff
@@ -138,7 +143,7 @@ data PackageIndex a = PackageIndex {
   --
   -- FIXME: Clarify what "preference order" means. Check that this invariant is
   -- preserved. See #1463 for discussion.
-  packageIdIndex :: !(Map PackageName (Map Version [a]))
+  packageIdIndex :: !(Map (PackageName, Maybe UnqualComponentName) (Map Version [a]))
```

Comment:
Oh wait that's what I meant write. My goal was eventually that prefiltering of the index that I mentioned on IRC, but that can be done with this too.